MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/javaservices/KotlinBorderedCircleActivity.kt

```diff
@@ -0,0 +1,294 @@
+    /**
+     * Update the [FillLayer] based on the GeoJSON retrieved via [getTurfPolygon].
+     *
+     * @param newCircleCenter the center coordinate to be used in the Turf calculation.
+     */
+    private fun drawPolygonCircle(newCircleCenter: Point) {
+        mapboxMap.getStyle { style ->
+            // Use Turf to calculate the coordinates for the circular-shaped Polygon
+            circlePolygonArea = getTurfPolygon(newCircleCenter, currentSetCircleRadius.toDouble())
+
+            // Set the source with the circle's GeoJSON
+            val polygonCircleSource = style.getSourceAs<GeoJsonSource>(CIRCLE_GEOJSON_SOURCE_ID)
+            polygonCircleSource?.setGeoJson(Polygon.fromOuterInner(
+                    LineString.fromLngLats(TurfMeta.coordAll(circlePolygonArea, false))))
+        }
+    }
+
+    /**
+     * Update the [FillLayer] based on the GeoJSON retrieved via [getTurfPolygon].
+     *
+     * @param newCircleCenter the center coordinate to be used in the Turf calculation.
+     */
+    private fun drawCircleBorder(newCircleCenter: Point) {
+        mapboxMap.getStyle {
+            // Use Turf to calculate the coordinates for the circular-shaped Polygon's border
+            val circleBorderPolygon = getTurfPolygon(newCircleCenter,
+                    currentSetCircleRadius.toDouble() - currentSetCircleBorderDifference)
+
+            // Set the source with the border's GeoJSON
+            val circleBorderSource = it.getSourceAs<GeoJsonSource>(CIRCLE_BORDER_GEOJSON_SOURCE_ID)
+            circleBorderSource?.setGeoJson(Polygon.fromOuterInner(
+                    // Create outer LineString
+                    LineString.fromLngLats(TurfMeta.coordAll(circleBorderPolygon, false)),
+                    // Create inter LineString
+                    LineString.fromLngLats(TurfMeta.coordAll(circlePolygonArea, false))
+            ))
+        }
+    }
+
+    /**
+     * Use the Turf library [TurfTransformation.circle] method to
+     * retrieve a [Polygon].
+     *
+     * @param centerPoint a [Point] which the circle will center around
+     * @param radius the radius of the circle
+     * @return a [Polygon] which represents the newly created circle
+     */
+    private fun getTurfPolygon(centerPoint: Point, radius: Double): Polygon {
+        return TurfTransformation.circle(centerPoint, radius, CIRCLE_STEPS, circleRadiusUnits)
+    }
+
+    override fun onStart() {
+        super.onStart()
+        mapView?.onStart()
+    }
+
+    public override fun onResume() {
+        super.onResume()
+        mapView?.onResume()
+    }
+
+    public override fun onPause() {
+        super.onPause()
+        mapView?.onPause()
+    }
+
+    override fun onLowMemory() {
+        super.onLowMemory()
+        mapView?.onLowMemory()
+    }
+
+    override fun onStop() {
+        super.onStop()
+        mapView?.onStop()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        mapboxMap.removeOnMapClickListener(this)
```

Comment:
The `private lateinit var mapboxMap: MapboxMap` could have not been initialised here. (e.g no network or the style is not loaded yet)

Reply:
Fixed. Thanks.